tests/cmdline/cmd_showbc_const.py.exp

```diff
@@ -0,0 +1,130 @@
+File cmdline/cmd_showbc_const.py, code block '<module>' (descriptor: \.\+, bytecode @\.\+ 159 bytes)
+Raw bytecode (code_info_size=34, bytecode_size=125):
+ 08 40 01 20 8c 0d 20 27 40 20 27 20 27 40 60 20
+ 27 24 40 60 40 24 27 47 24 27 67 40 27 47 27 47
+ 26 47 80 10 02 2a 01 1b 03 1c 02 16 02 59 11 07
+ 10 04 34 01 59 11 07 10 04 34 01 59 11 07 10 04
+ 34 01 59 11 07 10 04 34 01 59 42 42 42 35 23 00
+ 16 08 11 08 23 00 d9 44 47 11 07 10 04 34 01 59
+ 23 00 16 09 11 09 23 00 d9 44 47 11 07 10 04 34
+ 01 59 23 00 23 00 d9 44 47 11 07 10 04 34 01 59
+ 23 01 23 00 d9 44 47 11 07 10 05 34 01 59 50 23
+ 02 d9 44 47 11 07 10 04 34 01 59 42 40 51 63
+arg names:
+(N_STATE 2)
+(N_EXC_STACK 0)
+  bc=0 line=1
+  bc=0 line=2
+  bc=12 line=15
+  bc=12 line=16
+  bc=19 line=17
+  bc=19 line=19
+  bc=19 line=20
+  bc=26 line=21
+  bc=26 line=22
+  bc=33 line=23
+  bc=33 line=25
+  bc=33 line=28
+  bc=33 line=29
+  bc=40 line=30
+  bc=44 line=31
+  bc=44 line=33
+  bc=44 line=36
+  bc=44 line=38
+  bc=48 line=39
+  bc=55 line=40
+  bc=62 line=42
+  bc=66 line=43
+  bc=73 line=44
+  bc=80 line=47
+  bc=80 line=49
+  bc=87 line=50
+  bc=94 line=52
+  bc=101 line=53
+  bc=108 line=55
+  bc=114 line=56
+  bc=121 line=58
+00 LOAD_CONST_SMALL_INT 0
+01 LOAD_CONST_STRING 'const'
+03 BUILD_TUPLE 1
+05 IMPORT_NAME 'micropython'
+07 IMPORT_FROM 'const'
+09 STORE_NAME const
+11 POP_TOP
+12 LOAD_NAME print
+14 LOAD_CONST_STRING 'Kept'
+16 CALL_FUNCTION n=1 nkw=0
+18 POP_TOP
+19 LOAD_NAME print
+21 LOAD_CONST_STRING 'Kept'
+23 CALL_FUNCTION n=1 nkw=0
+25 POP_TOP
+26 LOAD_NAME print
+28 LOAD_CONST_STRING 'Kept'
+30 CALL_FUNCTION n=1 nkw=0
+32 POP_TOP
+33 LOAD_NAME print
+35 LOAD_CONST_STRING 'Kept'
+37 CALL_FUNCTION n=1 nkw=0
+39 POP_TOP
+40 JUMP 44
+42 JUMP 33
+44 LOAD_CONST_OBJ \.\+='foo'
```

Comment:
Are the jumps generated at +40 & +42 an (unrelated) bug with bytecode generation, or a limitation of the compiler?

Reply:
That's a limitation of the compiler, it doesn't eliminate all dead code, or follow unconditional jumps.